### PR TITLE
IGNITE-16349 Possible NPE Fix

### PR DESCRIPTION
**Pull request by vladErmakov07**

### CODE REVIEW

**Comment by zstan** [comment]
1.`
public Long2ObjectOpenHashMap<List<String>> remotes() {
        return remoteSources;
}
... 
do we need Nullable here ? or some check ?
public List<String> remotes(long exchangeId) {
        return fragmentDesc.remotes().get(exchangeId);
}
'

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/IgniteMdCollation.java [hunk 1]
```diff
@@ -342,7 +343,8 @@ private IgniteMdCollation() {
             fieldCollations.clear();
             boolean skip = false;
             for (RelFieldCollation ifc : ic.getFieldCollations()) {
-                final Collection<Integer> integers = targets.get(ifc.getFieldIndex());
+                final Collection<Integer> integers = targets.getOrDefault(ifc.getFieldIndex(),
+                        Collections.emptyList());
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/IgniteMdCollation.java:347]: 'targets' actually contains IntArrayList, isn't it?
```suggestion
                final IntArrayList integers = targets.getOrDefault(ifc.getFieldIndex(),
                        IntLists.emptyList());
```

Comment (AMashenkov) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/IgniteMdCollation.java:347]: fieldCollations.add(ifc.withFieldIndex(integers.iterator().**nextInt**()));

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/StopCalciteModuleTest.java [hunk 2]
```diff
@@ -167,6 +168,7 @@ public void cancel() {
         LOG.info(">>>> Starting test {}", testInfo.getTestMethod().orElseThrow().getName());
     }
 
+    @Disabled("IGNITE-16389")
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/StopCalciteModuleTest.java:171]: the mentioned ticket is resolved already

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java [hunk 3]
```diff
@@ -156,6 +156,7 @@ public ColocationGroup target() {
      * @param exchangeId ExchangeId to find remote nodes for.
      * @return Remote nodes for given exchangeId.
      */
+    @Nullable
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java:159]: if we annotated as Nullable let's mention in javadoc when null could be returned

Comment (vladErmakov07) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java:159]: Added mention in javadoc, thnx.

**Pull request closed without merging**